temporalio/contrib/openai_agents/_openai_runner.py

```diff
@@ -55,10 +56,21 @@ async def run(
                 )
 
         if starting_agent.mcp_servers:
-            raise ValueError(
-                "Temporal OpenAI agent does not support on demand MCP servers."
+            from temporalio.contrib.openai_agents import (
+                StatefulTemporalMCPServer,
+                StatelessTemporalMCPServer,
             )
 
+            for s in starting_agent.mcp_servers:
```

Comment:
I think we should full-blown error if the MCP server isn't an "MCP server ref" we provide and it doesn't match a name that was registered in the plugin.